modules/concept-docs/pages/n1ql-query.adoc

```diff
@@ -6,12 +6,22 @@
 [abstract]
 Parallel data management for complex queries over many records, using a familiar SQL-like syntax.
 
-The N1QL Query Language provides a familiar, SQL-like experience for querying documents stored in Couchbase.
-You can xref:6.5@server:n1ql:n1ql-language-reference/index.adoc[read up on the language in our reference guide], 
-but you probably just want to xref:howtos:n1ql-queries-with-sdk.adoc[dive into a practical example].
 
-Below, we fill in some of the gaps between reference and rolling-up-your-sleeves practicality,
-with discussion of a few areas of the Query service where more background knowledge will help you to better program your application.
+include::6.5@sdk:shared:partial$n1ql-queries.adoc[tag=intro]
+
+
+// Prepared Statements for Query Optimization
+include::6.5@sdk:shared:partial$n1ql-queries.adoc[tag=prepared]
+
+Do not turn off the `adhoc` flag for _every_ query to Server 6.0 and earlier, since only a finite number of query plans (currently 5000) can be stored in the SDK.
+
+[source,java]
+----
+QueryResult result = cluster.query(
+  "select count(*) from `travel-sample` where type = \"airports\" and country = ?",
```

Comment:
This is an invalid query. If you use ? , you also need to use a parameterized query to actually fill in the country